lib/krb5/cache.c

```diff
@@ -783,8 +839,14 @@ krb5_cc_configured_default_name(krb5_context context)
         return context->configured_default_cc_name;
 
 #ifdef _WIN32
-    if ((expanded = _krb5_get_default_cc_name_from_registry(context)))
-        return context->configured_default_cc_name = expanded;
+    if ((expanded = _krb5_get_default_cc_name_from_registry(context))) {
+        ret = resolve_cfg_default_name(context, expanded,
+                                       &context->configured_default_cc_name);
```

Comment:
The comment about asprintf() on line 811 is particularly important because resolve_cfg_default_name() is going to alter the value of `context->configured_default_cc_name` which could leave it non-NULL and pointing at garbage.